### PR TITLE
Path accept spaces

### LINKED ISSUE #174: Path with space are not identified as valid path
Due to the nature of our program, the `Path` object is made up of the Note's title.
The Note's title is not limited to alphanumeric but also spaces.

- [ ] Fix Regex to accept space for Path objects

### PR DESCRIPTION
**Pull request by ljiazh3ng**
resolves #174 
This PR changes the regex of the `Path` implementation to accept spaces.

### CODE REVIEW

File: src/main/java/com/notably/commons/core/path/RelativePath.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public String getStringRepresentation() {
 
     @Override
     public boolean equals(Object object) {
-        if (!(object instanceof Path)) {
+        if (!(object instanceof RelativePath)) {
             return false;
         }
         Path another = (Path) object;
```
Comment (firzanarmani) [inline src/main/java/com/notably/commons/core/path/RelativePath.java]: I feel like you can be more specific here and use `RelativePath another = (RelativePath) object`.

Likewise for `AbsolutePath` as well.

**Comment by firzanarmani** [comment]
I'm okay with this! However, due to the nature of the input string being trimmed, I will have to fix a bug that allows trailing (but not leading) whitespaces in the Title. I'll be fixing this is #188

**Review by firzanarmani — APPROVED** [review]
LGTM, but a small comment to consider 😄

**Comment by kevinputera** [comment]
@ljiazh3ng Actually, I wonder why we need to allow whitespace? It feels rather weird for me tbh to accept trailing and leading whitespace in our path components. Perhaps we'd wanna accept whitespace only if it's in the middle of a component?

For example, `/path /to / object` will be invalid, while `/this is a path/to object` is fine.

What do you think?

**Comment by firzanarmani** [comment]
> @ljiazh3ng Actually, I wonder why we need to allow whitespace? It feels rather weird for me tbh to accept trailing and leading whitespace in our path components. Perhaps we'd wanna accept whitespace only if it's in the middle of a component?
> 
> For example, `/path /to / object` will be invalid, while `/this is a path/to object` is fine.
> 
> What do you think?

@kevinputera, the way @ljiazh3ng designed it at this point; he allows leading and trailing spaces, then trims it. Since `Title` and `Path` components are the same, maybe they should follow the same validation?

File: src/main/java/com/notably/commons/core/path/AbsolutePath.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
  */
 public class AbsolutePath implements Path {
     public static final String INVALID_ABSOLUTE_PATH = "Invalid absolute path";
-    public static final String VALIDATION_REGEX = "\\/||(\\/\\p{Alnum}+)+\\/?";
+    public static final String VALIDATION_REGEX = "\\/|(\\/([a-zA-Z0-9]+\\s+)*[a-zA-Z0-9]+)*\\/?";
```
Comment (kevinputera) [inline src/main/java/com/notably/commons/core/path/AbsolutePath.java:14]: @ljiazh3ng this first part `\\/` is actually unnecessary anymore.
The regex can be `(\\/(\\p{Alnum}+\\s+)*\\p{Alnum}+)*\\/?` only!

**Review by kevinputera — APPROVED** [review]
Looks great!

**Pull request merged**